### ISSUE TITLE
offline install - "future" error

### ISSUE THREAD

**Issue by stweed** [pyinstaller/pyinstaller#2493]
I'm trying to install pyinstaller on windows 7 without internet access. I've downloaded the archive, un-zipped it and run the setup.py. Everything looked great until I got this error "Error: could not find suitable distribution for Requirement.parse('future')" It seems that pyinstaller is looking for the website address "https://pypi.python.org/simple/future/" but of course can not find it. Any way to work around this, or is this a bug?

Sorry if I'm not too familiar with this sort of thing, I'm kind of new to running programs from the commandline as well as python in general.

- Spencer
![pyistaller capture](https://cloud.githubusercontent.com/assets/13682597/23767472/833fc806-04bd-11e7-9a55-ff48db8916ae.PNG)

**Comment by htgoebel** [comment]
When installing on a offline system, you need to resolve dependencies manually - like with any other Python package.

**Comment by stweed** [comment]
Great.  Can you point me in the right direction to figure that out? I've got no problem reading documentation, just a helpful pointer would be appreciated, I'm pretty new to python as you can probably tell.

**Comment by htgoebel** [comment]
pip download

**Comment by stweed** [comment]
Thanks man, super helpful. So when I run "pip install pyinstaller" it tells me "collecting future (from pyinstaller)" and then just errors out on "getaddrinfo failed". As stated above what I think I'm running into is that pyinstaller is looking online for whatever 'future' is. Does anyone know how I can perhaps download this locally and tell pip where it is? Again sorry that I'm not more familiar with this - I'm perfectly happy studying documentation but there doesn't seem to be a lot out there to help me out on this particular situation...

**Comment by tallforasmurf** [comment]
That was a bit abrupt by @htgoebel, wasn't it? I think he feels impatient when people use github issues for help queries instead of using [the mailing list](https://groups.google.com/forum/#!forum/PyInstaller). I also was not sure what he meant at first, but I think I do understand now I have looked at the [pip docs for download](https://pip.pypa.io/en/stable/reference/pip_download/), which I had never heard of before.

It appears you could use `pip download pyinstaller` and it will download and save all dependencies of that package. I ran it and it looked like this:

~~~~~
 pip download pyinstaller
Collecting pyinstaller
  Downloading PyInstaller-3.2.1.tar.bz2 (2.4MB)
    100% |████████████████████████████████| 2.4MB 552kB/s 
  Saved ./PyInstaller-3.2.1.tar.bz2
Collecting setuptools (from pyinstaller)
  Downloading setuptools-34.3.1-py2.py3-none-any.whl (389kB)
    100% |████████████████████████████████| 399kB 1.9MB/s 
  Saved ./setuptools-34.3.1-py2.py3-none-any.whl
Collecting six>=1.6.0 (from setuptools->pyinstaller)
  Downloading six-1.10.0-py2.py3-none-any.whl
  Saved ./six-1.10.0-py2.py3-none-any.whl
Collecting appdirs>=1.4.0 (from setuptools->pyinstaller)
  Downloading appdirs-1.4.3-py2.py3-none-any.whl
  Saved ./appdirs-1.4.3-py2.py3-none-any.whl
Collecting packaging>=16.8 (from setuptools->pyinstaller)
  Downloading packaging-16.8-py2.py3-none-any.whl
  Saved ./packaging-16.8-py2.py3-none-any.whl
Collecting pyparsing (from packaging>=16.8->setuptools->pyinstaller)
  Downloading pyparsing-2.2.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 4.7MB/s 
  Saved ./pyparsing-2.2.0-py2.py3-none-any.whl
Successfully downloaded pyinstaller setuptools six appdirs packaging pyparsing
~~~~~

I don't see anything in there about "future" but maybe it is included in one of the others, or you could use `pip download future` just to be sure, before you sneakernet all the above `.whl` files over the disconnected machine.

pyinstaller recommends using pip-Win and virtualenv as [discussed here](https://www.dabapps.com/blog/introduction-to-pip-and-virtualenv-python/), this would be true of both the connected and unconnected machines.

**Comment by stweed** [comment]
Thanks for the help tallforasmurf, I'll check it out. I did feel funny posting a question on github but I couldn't find support on the pyinstaller website and this seemed like the next most likely candidate when I found someone else had posted a similar question a while back. Unfortunately our internet stations are also pretty locked down and I can't install pip without an admin, so I'll have to do this some time later at home... Anyways I'll continue any future questions on the mailing list as you suggest!
